lib/matplotlib/colorbar.py

```diff
@@ -501,17 +499,23 @@ def draw_all(self):
         # sets self._boundaries and self._values in real data units.
         # takes into account extend values:
         self._process_values()
-        # sets self.vmin and vmax in data units, but just for
-        # the part of the colorbar that is not part of the extend
-        # patch:
+        # sets self.vmin and vmax in data units, but just for the part of the
+        # colorbar that is not part of the extend patch:
         self._find_range()
-        # returns the X and Y mesh, *but* this was/is in normalized
-        # units:
+        # returns the X and Y mesh, *but* this was/is in normalized units:
         X, Y = self._mesh()
         C = self._values[:, np.newaxis]
-        # decide minor/major axis
-        self._config_axis()
-        self._config_axes(X, Y)
+
+        self._config_axis()  # Inline it after deprecation elapses.
+        # Configure axes limits, patch, and outline.
+        xy = self._outline(X, Y)
+        xmin, ymin = xy.min(axis=0)
+        xmax, ymax = xy.max(axis=0)
+        self.ax.set(xlim=(xmin, xmax), ylim=(ymin, ymax))
```

Comment:
Is this really the same, since it doesn't update `self.ax.dataLim` any more?

Reply:
If someone really consults the colorbar axes' datalim then no it's not, but in practice the datalim are only used (should only be used...) for computing autoscaled limits, and we're not going to rely on autoscaling the colorbar axes anyways.